src/components/TodoInfo/TodoInfo.tsx

```diff
@@ -0,0 +1,142 @@
+import {
+  useState,
+  FC,
+  ChangeEvent,
+  useEffect,
+  useRef,
+  memo,
+  KeyboardEvent,
+} from 'react';
+import cn from 'classnames';
+import { TodoInfoProps } from './TodoInfoProps';
+
+export const TodoInfo: FC<TodoInfoProps> = memo(({
+  todo,
+  removeTodos,
+  loadingTodoIds,
+  handleUpdate,
+}) => {
+  const {
+    title,
+    completed,
+    id,
+  } = todo;
+  const formRef = useRef<HTMLInputElement | null>(null);
+  const [todoTitle, setTodoTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+
+  useEffect(() => {
+    if (formRef.current) {
+      formRef.current.focus();
+    }
+
+    const updatedTitle = title.trim().replace(/\s+/g, ' ');
+
+    setTodoTitle(updatedTitle);
+  }, [title, isEditing]);
+
+  const handleQueryChange = (
+    event: ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = () => {
+    const updatedTitle = title.trim().replace(/\s+/g, ' ');
+    const updatedQuery = todoTitle.trim().replace(/\s+/g, ' ');
+
+    if (updatedTitle === updatedQuery) {
+      setTodoTitle(updatedQuery);
+      setIsEditing(false);
+
+      return;
+    }
+
+    if (!updatedQuery) {
+      removeTodos([id]);
+      setTodoTitle(updatedTitle);
+      setIsEditing(false);
+
+      return;
+    }
+
+    handleUpdate([id], todoTitle);
+    setIsEditing(false);
+  };
+
+  const handleOnClickTodoStatus = () => handleUpdate([id]);
+  const handleOnDoubleClickTitle = () => setIsEditing(true);
+  const handleOnClickRemoveTodo = () => removeTodos([id]);
+  const handleOnKeyUpEscapeTitle = (
+    event: KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Escape') {
```

Comment:
It would be cool not to use magic values and put such strings into constants and reuse them:
```
const ESCAPE = 'Escape';
```
This will increase the readability of the code and save us from possible errors.
This applies to development in general.